app/src/main/java/org/mifos/mobile/ui/activities/LoginActivity.kt

```diff
@@ -88,6 +89,7 @@ class LoginActivity : BaseActivity(), LoginView {
      * Starts [PassCodeActivity]
      */
     override fun showPassCodeActivity() {
+        EspressoIdlingResouce.decrement()
```

Comment:
I don't think, for writing integration tests or be it unit test we will have to make changes in the testing classes, check on this.

Reply:
Using `EspressoIdlingResource` demands production-level code be altered for letting Espresso know that the UI is being updated due to a task. This is preferred over sleeping the thread in the test resource. More information is available in the [documentation](https://developer.android.com/reference/androidx/test/espresso/idling/CountingIdlingResource/)

Reply:
Are we making an actual API call, during the Integration test? If yes, let's not do it.
We can mock the API call, with some hardcoded JSONs, check on the feasibility of the solution which I'm referring to in this comment?

Reply:
I will check for the feasibility and update on it sooner. Meanwhile, I will mark this PR as a draft